hydromt_delft3dfm/dflowfm.py

```diff
@@ -1487,6 +1458,385 @@ def setup_1dboundary(
             da_out, name=f"boundary1d_{da_out.name}_{branch_type}"
         )  # FIXME: this format cannot be read back due to lack of branch type info from model files
 
+    def _read_forcing_geodataset(
+        self, forcing_geodataset_fn: Union[str, Path], forcing_name: str = "discharge"
+    ):
+        """Read forcing geodataset."""
+
+        refdate, tstart, tstop = self.get_model_time()  # time slice
+
+        if (
+            forcing_geodataset_fn is not None
+            and self.data_catalog[forcing_geodataset_fn].data_type == "GeoDataset"
+        ):
+            da = self.data_catalog.get_geodataset(
+                forcing_geodataset_fn,
+                geom=self.region.buffer(
+                    1000
```

Comment:
In some of your functions a snap_offset is used. Maybe you could this value or a multiplier here instead of fixed 1000? (I have to double checked if buffer value is always meters even if CRS is in degrees)

Reply:
Yes, the snap_offset can indeed be handy here.